crates/sargon-uniffi/src/signing/transaction_intent/hd_signature_input.rs

```diff
@@ -0,0 +1,45 @@
+use crate::prelude::*;
+use sargon::TransactionIntentHash as InternalTransactionIntentHash;
+type InternalHDSignatureInputForTransactionIntent =
+    sargon::HDSignatureInput<InternalTransactionIntentHash>;
+
+#[derive(Clone, PartialEq, Eq, Hash, uniffi::Record)]
+pub struct HDSignatureInputForTransactionIntent {
```

Comment:
macro for this and the otherone.

When translate from Generics in `sargon` -> `sargon-uniffi` (oh nooes, no generics) then macros are good to ensure the implementations are "in sync".